index.d.ts

```diff
@@ -0,0 +1,130 @@
+declare module 'fluture' {
+  interface ResolveFunction<T> {
+    (value: T): void
+  }
+  interface RejectFunction {
+    (error: Error): void
+  }
+
+  interface FunctionReturning<T extends Function> {
+    (...t: any[]): T
+  }
+  abstract class Future<T> {
+    done<T> (fn: (err: Error, value: T) => void): Future<T>
+  }
+
+  function after<T> (duration: number, value: string): Future<T>
+
+  function and<A, B> (a: Future<A>, b: Future<B>): Future<A | B>
```

Comment:
The return type can be specialized to be `Future<B>` (the resolution value of `a` is always ignored), when we introduce the error branch I think we can further improve the signature to:

```ts
function and<L, R> (a: Future<L, any>, b: Future<L, R>): Future<L, R>
```